src/world_interface/kinematic_common_interface.cpp

```diff
@@ -162,12 +162,11 @@ double getJointPowerValue(types::jointid_t joint) {
 
 void setJointMotorPower(robot::types::jointid_t joint, double power) {
 	using robot::types::jointid_t;
-	if (Constants::arm::JOINT_MOTOR_MAP.find(joint) != Constants::arm::JOINT_MOTOR_MAP.end()) {
-		// TODO: check if associated motor is in IK_MOTORS, don't check specifically these ones
-		if (!Globals::armIKEnabled ||
-			!std::find(Constants::arm::IK_MOTOR_JOINTS.begin(),
-					   Constants::arm::IK_MOTOR_JOINTS.end(), joint)) {
-			setMotorPower(Constants::arm::JOINT_MOTOR_MAP.at(joint), power);
+	if (Constants::JOINT_MOTOR_MAP.find(joint) != Constants::JOINT_MOTOR_MAP.end()) {
+		bool isIKMotor = std::find(Constants::arm::IK_MOTOR_JOINTS.begin(),
+								   Constants::arm::IK_MOTOR_JOINTS.end(), joint);
```

Comment:
I definitely lied to you earlier whoops, `std::find` doesn't return a `bool`, it returns an iterator. You should check `bool isIKMotor = std::find(...) == IK_MOTORS_JOINTS.end()`.

Reply:
@abhaybd `bool isIKMotor = std::find(...) == IK_MOTORS_JOINTS.end()` will return `true` if it doesn't find a motor. I think you meant `bool isIKMotor = std::find(..) != IK_MOTORS_JOINTS.end()`

Reply:
Whoops yeah you're right, @geeoon do that instead